src/osd/PG.cc

```diff
@@ -2713,7 +2714,8 @@ std::pair<ghobject_t, bool> PG::do_delete_work(
   bool running = true;
   if (num) {
     dout(20) << __func__ << " deleting " << num << " objects" << dendl;
-    Context *fin = new C_DeleteMore(this, get_osdmap_epoch());
+    int64_t cost = num * cct->_conf->osd_pg_delete_cost;
```

Comment:
Instead of calculating the 'cost' here, `num` could be passed and set in C_DeleteMore context and eventually to `queue_for_pg_delete()' which can use `num` to calculate the overall cost in the lambda function I suggested in my earlier comment. This way the cost calculation is localized to `queue_for_pg_delete()` .

Reply:
Yeah that would be neater, would it be okay to pass num as 1 in src/osd/PrimaryLogPG.cc https://github.com/ceph/ceph/blob/main/src/osd/PrimaryLogPG.cc#L12853?

Reply:
Yes, it should be fine to pass `num` as 1 along with a comment mentioning why.